Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -60,14 +60,14 @@ def ensure_running(self):
                 fds_to_pass.append(sys.stderr.fileno())
             except Exception:
                 pass
-            cmd = 'from multiprocessing.semaphore_tracker import main;main(%d)'
+            cmd = 'from multiprocessing.semaphore_tracker import main;main({}, {})'
             r, w = os.pipe()
             try:
                 fds_to_pass.append(r)
                 # process will out live us, so no need to wait on pid
                 exe = spawn.get_executable()
                 args = [exe] + util._args_from_interpreter_flags()
-                args += ['-c', cmd % r]
+                args += ['-c', cmd.format(r, sys.stderr.fileno())]
```

Comment:
This won't work if `sys.stderr` isn't an actual file:
```
>>> io.StringIO().fileno()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
io.UnsupportedOperation: fileno
```

Reply:
I was relaying on the fact that the tracker seems to work under the assumption that `sys.stderr` has a file descriptor associated:

https://github.com/python/cpython/blob/master/Lib/multiprocessing/semaphore_tracker.py#L60

Reply:
@pitrou I am missing something?

Reply:
Well, the `except Exception` should be clear, no? :-)

Reply:
Ups, my bad. :)

Let me investigate what options do we have. Do you have a preferred approach on how to handle this?